internal/graph/check.go

```diff
@@ -196,8 +196,7 @@ func WithMaxConcurrentReads(limit uint32) LocalCheckerOption {
 // NewLocalChecker constructs a LocalChecker that can be used to evaluate a Check
 // request locally.
 //
-// The constructed LocalChecker is not wrapped with cycle detection. Developers
-// wanting a LocalChecker without other wrapped layers (e.g caching and others)
+// Developers wanting a LocalChecker with other optional layers (e.g caching and others)
```

Comment:
The "optional" word is key here. It would have been easy to do `NewLocalChecker` elsewhere in the codebase (e.g. for ListObjects, or in a test) and forget that it doesn't include cycle detection.